src/main/java/calculator/domain/OperationUnit.java

```diff
@@ -0,0 +1,69 @@
+package calculator.domain;
+
+public class OperationUnit {
+    private Operand left;
+    private Operator operator;
+    private Operand right;
+
+    public OperationUnit() {
+    }
+
+    public OperationUnit(int left, String symbol, int right) {
+        this.left = new Operand(left);
+        this.operator = Operator.getOperatorBySymbol(symbol);
+        this.right = new Operand(right);
+    }
+
+    public void add(String element) {
+        if (left == null) {
+            left = new Operand(element);
+            return;
+        }
+
+        if (operator == null) {
+            operator = Operator.getOperatorBySymbol(element);
+            return;
+        }
+
+        if (right == null) {
+            right = new Operand(element);
+            return;
+        }
+
+        throw new IllegalStateException("연산 단위가 이미 완성되었습니다.");
+    }
+
+    public boolean isCalculable() {
+        return left != null && operator != null && right != null;
+    }
+
+    public void calculate() {
+        if (!isCalculable()) {
+            throw new IllegalStateException("연산 단위가 완성되지 않았습니다.");
+        }
+
+        int left = this.left.getValue();
+        int right = this.right.getValue();
+
+        int result = operator.calculate(left, right);
+        clear();
+        passPreviousResult(result);
+    }
```

Comment:
큐를 활용한다면, clear 와 같은 메서드는 필요 없을것 같기도하네요!
자료구조를 활용해보는것도 직관적일것 같아요 :)

```java
    private final Deque<Operand> operands = new ArrayDeque<>();
    private final Deque<Operator> operators = new ArrayDeque<>();
```

```suggestion
    public void calculate() {
        if (!isCalculable()) {
            throw new IllegalStateException("연산 단위가 완성되지 않았습니다.");
        }

        Operand left = operands.pollFirst();
        Operand right = operands.pollFirst();
        Operator operator = operators.pollFirst();

        int result = operator.calculate(left.getValue(), right.getValue());
        operands.addFirst(new Operand(result));
    }
```

적용할 필욘 없고, 이런 방법도 있을것 같다라는 의견입니다 :)